html5/js/ImageDecoder.js

```diff
@@ -7,62 +7,37 @@
  *
  */
 
-/*
- * Receives native image packets and decode them via ImageDecoder.
- * https://developer.mozilla.org/en-US/docs/Web/API/ImageDecoder
- * ImageDecoder is only working in Chrome 94+ and Android
- *
- */
-
-const XpraImageDecoderLoader = {
-  hasNativeDecoder() {
-    return typeof ImageDecoder !== "undefined";
-  },
-};
-
 class XpraImageDecoder {
-  constructor() {
-    this.on_frame_decoded = null;
-    this.on_frame_error = (packet, error) => {
-      console.error("ImageDecoder error on packet", packet, ":", error);
-    };
-  }
 
-  queue_frame = function (packet) {
+  async convertToBitmap(packet) {
```

Comment:
If I understand this correctly, this relies on the fact that the browser will process calls to `convertToBitmap` (and the underlying decoding functions that this function calls) in sequential order.
If at some point this changes (perhaps they introduce multi-core image decoding threads? or a codec ends up delegated to another thread because of GPU or whatnot) then this may break and return the bitmaps in an order which is different from the one we submitted them in, which will cause image tearing and visual artefacts.
If so, is there anything we can do to prevent that?
(perhaps somehow tie this to `requestAnimationFrame` vsync?)

Reply:
So, this comment was wrong: we actually block here with `async` and it is impossible to end up decoding out of order.